lib/Doctrine/ODM/PHPCR/Tools/Helper/TranslationConverter.php

```diff
@@ -0,0 +1,228 @@
+use Doctrine\ODM\PHPCR\Translation\TranslationStrategy\ChildTranslationStrategy;
+use Doctrine\ODM\PHPCR\Translation\TranslationStrategy\NonTranslatedStrategy;
+use Doctrine\ODM\PHPCR\Translation\TranslationStrategy\TranslationStrategyInterface;
+use Doctrine\ODM\PHPCR\DocumentManagerInterface;
+use Doctrine\ODM\PHPCR\Mapping\ClassMetadata;
+
+/**
+ * Migrate a field that has become translated or changed its translation
+ * strategy.
+ *
+ * To avoid problems with large repositories, the converter is operating in a
+ * batched mode. Call it repeatedly and call flush and clear between calls.
+ *
+ * You need to call `save()` on the PHPCR session, rather than `flush()` on
+ * the document maanger. To avoid unexpected results, it is recommended to
+ * `clear()` the document manager before starting to convert and after having
+ * saved the PHPCR session.
+ *
+ * @author David Buchmann <mail@davidbu.ch>
+ */
+class TranslationConverter
+{
+    /**
+     * @var DocumentManagerInterface
+     */
+    private $dm;
+
+    /**
+     * @var int Number of documents to process per batch.
+     */
+    private $batchSize;
+
+    /**
+     * @param DocumentManagerInterface $dm
+     * @param int                      $batchSize
+     */
+    public function __construct(DocumentManagerInterface $dm, $batchSize = 200)
+    {
+        $this->dm = $dm;
+        $this->batchSize = $batchSize;
+    }
+
+    /**
+     * Migrate content into the new translation form and remove the old properties.
+     *
+     * This does not commit the changes to the repository. Call save on the
+     * PHPCR session *after each batch*. Calling flush on the document manager
+     * *is not enough*.
+     *
+     * When un-translating, the current locale and language fallback is used.
+     * When translating, the new properties are copied into all languages.
+     *
+     * To convert all fields into a translation, you can pass an empty array
+     * for $fields and the information is read from the metadata. The fields
+     * are mandatory when converting fields back to non-translated.
+     *
+     * To convert a single field from translated to non-translated, simply
+     * specify that field.
+     *
+     * @param string       $class                FQN of the document class
+     * @param string|null  $previousStrategyName Name of previous strategy or null if field was not
+     *                                           previously translated
+     * @param array        $fields               List of fields to convert. Required when making a
+     *                                           field not translated anymore
+     *
+     * @return boolean true if there are more documents to convert and this method needs to be
+     *                      called again.
+     *
+     * @throws PHPCRExceptionInterface if the document can not be found.
+     */
+    public function convert(
+        $class,
+        $previousStrategyName = null,
```

Comment:
you understood correctly. i updated the docblock to be explicit.

regarding constants: i agree that we should provide constants for the default strategies. currently, their names are simply written in https://github.com/doctrine/phpcr-odm/blob/translation-conversion/lib/Doctrine/ODM/PHPCR/DocumentManager.php#L129-L130

if you want to do a pull request to provide the constants and replace string occurences with the constants, please do. not sure where the best location is though - the interface seems a bit wrong, maybe each strategy implementation? when you use the constant you do know about the strategy...